openapi_python_client/parser/openapi.py

```diff
@@ -88,15 +90,16 @@ class Endpoint:
     tag: str
     summary: Optional[str] = ""
     relative_imports: Set[str] = field(default_factory=set)
-    query_parameters: List[Property] = field(default_factory=list)
-    path_parameters: List[Property] = field(default_factory=list)
-    header_parameters: List[Property] = field(default_factory=list)
-    cookie_parameters: List[Property] = field(default_factory=list)
+    query_parameters: Dict[str, Property] = field(default_factory=dict)
+    path_parameters: Dict[str, Property] = field(default_factory=dict)
+    header_parameters: Dict[str, Property] = field(default_factory=dict)
+    cookie_parameters: Dict[str, Property] = field(default_factory=dict)
     responses: List[Response] = field(default_factory=list)
     form_body_class: Optional[Class] = None
     json_body: Optional[Property] = None
     multipart_body: Optional[Property] = None
     errors: List[ParseError] = field(default_factory=list)
+    used_python_identifiers: Set[PythonIdentifier] = field(default_factory=set)
```

Comment:
I see where this is written, but not where it's read. Is this residual or is there a missing piece?

Reply:
Added usage.